MonoGame.Framework/Graphics/Effect/Effect.cs

```diff
@@ -122,15 +124,14 @@ public Effect (GraphicsDevice graphicsDevice, byte[] effectCode)
             Clone(cloneSource);
         }
 
-        private MGFXHeader ReadHeader(byte[] effectCode)
+        private MGFXHeader ReadHeader(byte[] effectCode, int index = 0)
```

Comment:
This is an internal API...  I would remove the default parameter value and make everyone deal with the new argument.